Assets/CesiumForUnitySamples/Scripts/CesiumSamplesMetadataPickingVR.cs

```diff
@@ -0,0 +1,137 @@
+    "name:es",
+    "name:etymology",
+    "name:fr",
+    "name:he",
+    "name:hi",
+    "name:hr",
+    "name:ja",
+    "name:ko",
+    "name:pl",
+    "name:uk",
+    "name:zh",
+    "name:zh_pinyin",
+    "nycdoitt:bin",
+    "part#elementId",
+    "part#elementType",
+    "part#building:colour",
+    "part#roof:colour",
+    "part#roof:direction",
+    "part#addr:city",
+    "part#addr:country",
+    "part#addr:housenumber",
+    "part#addr:postcode",
+    "part#addr:state",
+    "part#addr:street",
+    "part#wikidata",
+    "part#nycdoitt:bin",
+    "roof:colour",
+    "type",
+    "wikidata",
+  };
+
+    string GetInterestingProperties(CesiumFeature feature)
+    {
+        string ret = "";
+        foreach (var propertyName in feature.properties)
+        {
+            if (!ignoreProperties.Contains(propertyName))
+            {
+                string propertyValue = feature.GetString(propertyName, "null");
+                if (propertyValue != "null")
+                {
+                    ret += $"{propertyName}: {propertyValue}\n";
+                }
+            }
+        }
+        return ret;
+    }
+
+    protected override void OnActivated(ActivateEventArgs args)
+    {
+        RaycastHit hit;
+        Transform interactorTransform = args.interactorObject.transform;
+        if (Physics.Raycast(interactorTransform.position,
+                            interactorTransform.TransformDirection(Vector3.forward),
+                            out hit, Mathf.Infinity))
+        {
+            CesiumMetadata metadata =
+                hit.transform.GetComponentInParent<CesiumMetadata>();
+            if (metadata != null)
+            {
+                CesiumFeature[] features =
+                    metadata.GetFeatures(hit.transform, hit.triangleIndex);
+                if (features.Length > 0)
+                {
+                    CesiumFeature feature = features[0];
+                    string name = feature.GetString("name", null);
+                    if (name == "null")
+                    {
+                        name = "Name N/A";
+                    }
+                    metadataText.text =
+                        $"<size=150%><b>{name}</b></size>\n<size=75%>{GetInterestingProperties(feature)}</size>";
+                }
+            }
+        }
+    }
+}
+public class CesiumSamplesMetadataPickingVR : MonoBehaviour
+{
+    public Cesium3DTileset tileset;
+    public CharacterController characterController;
+    public TextMeshProUGUI metadataText;
+    void Start()
+    {
+        if (tileset != null && characterController != null &&
+            metadataText != null)
+        {
+            tileset.OnTileGameObjectCreated += go => {
+                foreach (Transform child in go.transform)
+                {
+                    var mc = child.GetComponent<MeshCollider>();
+                    if (mc != null)
+                    {
+                        Physics.IgnoreCollision(mc, characterController);
```

Comment:
Is this necessary for the level to work? Or was this just a decision to let the player pass through buildings?

Reply:
Yes, you would start walking on top of the smaller buildings. Also I thought it was neat to demonstrate customizing the game objects.